pilot/model/validation.go

```diff
@@ -964,9 +964,9 @@ func ValidateEgressRuleDomain(domain string) error {
 }
 
 // ValidateEgressRulePort checks the port of the egress rule (communication port and protocol)
-func ValidateEgressRulePort(port *routing.EgressRule_Port) error {
+func ValidateEgressRulePort(port *routing.Port) error {
 
-	if err := ValidatePort(int(port.Port)); err != nil {
+	if err := ValidatePort(int(port.GetNumber())); err != nil {
 		return err
 	}
 
```

Comment:
You need a conversion function here, to convert the Port spec into our internal Port struct in the service model. That would allow you to reuse code in proxy/envoy/